src/box/recovery.cc

```diff
@@ -528,6 +528,8 @@ recovery_follow_local(struct recovery *r, struct xstream *stream,
 	 */
 	assert(r->watcher == NULL);
 	r->watcher = fiber_new_system(name, hot_standby_f);
+	if (r->watcher == NULL)
+		diag_raise();
```

Comment:
I'm not sure if it's possible to raise here… There are places when we panic in such cases:
https://github.com/tarantool/tarantool/blob/9549310d15dd16c6fd49cc658bdf9fffb82ba63c/src/box/gc.c#L121-L128

https://github.com/tarantool/tarantool/blob/9549310d15dd16c6fd49cc658bdf9fffb82ba63c/src/box/watcher.c#L121-L125

https://github.com/tarantool/tarantool/blob/9549310d15dd16c6fd49cc658bdf9fffb82ba63c/src/box/vy_log.c#L770-L773

https://github.com/tarantool/tarantool/blob/9549310d15dd16c6fd49cc658bdf9fffb82ba63c/src/box/vy_scheduler.c#L437-L440

https://github.com/tarantool/tarantool/blob/9549310d15dd16c6fd49cc658bdf9fffb82ba63c/src/box/raft.c#L159-L173

Reply:
I followed the error handling pattern of the surrounding code.

We panic at the exception landing pad anyways:
https://github.com/tarantool/tarantool/blob/a8c6c27cad118b49ec795223edb620022a4334a5/src/box/box.cc#L4850-L4859